app/bundles/CoreBundle/Service/BatchDeleteService.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Mautic\CoreBundle\Service;
+
+use Mautic\CoreBundle\Entity\CommonEntity;
+use Mautic\CoreBundle\Model\FormModel;
+use Mautic\CoreBundle\Model\MauticModelInterface;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\LeadBundle\Entity\Tag;
+
+final class BatchDeleteService
+{
+    public const LOAD_RESULTS_IN_CHUNKS_OF = 200;
```

Comment:
I'd change the batch to 100 max. Otherwise it's good that batches are used 👍

Reply:
Reduced the batch size to 100